### PR TITLE
fix: removes request parsing from "onUnhandledRequest"

### LINKED ISSUE #1983: Drop request parsing from "onUnhandledRequest"
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

Right now, we parse _any_ unhandled request just to see if it's not a GraphQL request. We do that to provide a bit better developer experience by including the operation type and name in the unhandled request warning message. 

However, because of this, everybody ends up with all GraphQL dependencies even if they are not mocking anything GraphQL related. This has also been reported to end up with a lot of additional modules imported when using MSW (by @thepassle). 

## Solution

- Remove request parsing from `onUnhandledRequest` at all. 
- Log out request method and path. This is sufficient to help the developer debug what's unmatched. 
- In the future, we can log out the request initiator, once we implement it (#1764)

### PR DESCRIPTION
**Pull request by kettanaito**
- Closes #1983 

This isn't a breaking change because the behavior modified is not connected to what MSW promises the developer. This is a warning message change, basically. But it has huge performance implications for everyone:

- MSW no longer imports `graphql` for everyone even if they aren't using the `graphql` handler;
- MSW no longer does request parsing for unhandled requests just to see if they weren't GraphQL requests. This brings significant performance boost in general and even bigger so for teams with a large number of request handlers.

### CODE REVIEW

**Review by mattcosta7 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -113,21 +113,18 @@
...
     "@types/cookie": "^0.6.0",
-    "@types/js-levenshtein": "^1.1.3",
     "@types/statuses": "^2.0.4",
     "chalk": "^4.1.2",
     "chokidar": "^3.4.2",
     "graphql": "^16.8.1",
     "headers-polyfill": "^4.0.2",
     "inquirer": "^8.2.0",
     "is-node-process": "^1.2.0",
-    "js-levenshtein": "^1.1.6",
```
Comment (mattcosta7) [inline package.json:130]: 🚀

**Comment by thepassle** [comment]
Module graph on `main` branch:
image

Module graph on this branch:
image

Nice improvement :)

**Comment by kettanaito** [comment]
@thepassle, that is _amazing_. All thanks to @mattcosta7 and you for pointing these things out. Let's have this released!

**Comment by thepassle** [comment]
> @thepassle, that is _amazing_

Especially consider that we went from:
image

To:
image

**Comment by kettanaito** [comment]
## Released: v2.1.4 🎉

This has been released in v2.1.4!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.1.4)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.1.4)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**